### PR TITLE
refactor(decorators/middlewares): rename to `UseMiddlewares` and mark old one as deprecated

### PR DESCRIPTION
**Pull request by Mnigos**
Sadly I cannot check if it works, because I'm getting `Cannot find module 'nestjs-trpc' or its corresponding type declarations.` error inside examples.

I also changed every mention about `@Middlewares` decorator inside documentation and examples to use newer version.

### CODE REVIEW

**Review by KevinEdry — COMMENTED** [review]
Thanks for opening a PR to change the decorator name, it looks good overall but I think the adapter is also checking for a specific decorator name either in the factory or the generator section, i'll check it later today.

**Comment by Mnigos** [comment]
> Thanks for opening a PR to change the decorator name, it looks good overall but I think the adapter is also checking for a specific decorator name either in the factory or the generator section, i'll check it later today.

Sure!

**Comment by Mnigos** [comment]
> Thanks for opening a PR to change the decorator name, it looks good overall but I think the adapter is also checking for a specific decorator name either in the factory or the generator section, i'll check it later today.

If you meant this lines inside `decorator.generator.ts`
```ts
else if (decoratorName === 'UseMiddlewares') {
          return array;
        }
```

Then it's changed already. I searched whole repository for any `Middlewares` mention and I think I covered everything, but for sure it will be better if you will check it yourself.

**Comment by KevinEdry** [comment]
Theres an enum somewhere, I'll go over it and test it.

File: packages/nestjs-trpc/lib/generators/decorator.generator.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class DecoratorGenerator {
               ...(output ? { output } : {}),
             },
           });
-        } else if (decoratorName === 'Middlewares') {
+        } else if (decoratorName === 'UseMiddlewares') {
```
Comment (KevinEdry) [inline packages/nestjs-trpc/lib/generators/decorator.generator.ts:50]: This will disable schema generation for the `Middleware` decorator.

**Review by KevinEdry — CHANGES_REQUESTED** [review]
I haven't tested it yet, but the changes will disable the `Middleware` decorator which is undesirable.

File: packages/nestjs-trpc/lib/generators/decorator.generator.ts (hunk 1 above)
Comment (Mnigos) [reply packages/nestjs-trpc/lib/generators/decorator.generator.ts]: Should be fixed now.

**Review by KevinEdry — APPROVED** [review]
I've tested it and it seems to work, approved!

**Pull request merged**